benchmarks/sha2.cc

```diff
@@ -159,7 +174,7 @@ BENCHMARK_CAPTURE(HACL_Sha2_oneshot,
                   sha2_224,
                   HACL_HASH_SHA2_224_DIGEST_LENGTH,
                   expected_digest_sha2_224,
-                  Hacl_Hash_SHA2_hash_224)
+                  Hacl_Streaming_SHA2_sha224)
```

Comment:
This makes the benchmark pretend like the oneshot function exposed by the library have this performance. Which they don't
I agree that this should be used but a few things need to happen as well
* remove the slow API
* update the documentation to point consumers to the right API

I'm fine with doing this in follow-ups. But it either needs to be filed as follow-ups to tackle after this (maybe @duesee or @pnmadelaine can take care of this), or done here.